src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -642,6 +642,35 @@ public void SetConsoleUICulture()
             thisThread.CurrentUICulture = originalUICulture;
         }
 
+
+        [Fact]
+        public void ConsoleUIRespectsSDKLanguage()
+        {
+            const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
+
+            // Save the current environment info so it can be restored.
+            var originalUILanguage = Environment.GetEnvironmentVariable(DOTNET_CLI_UI_LANGUAGE);
+            Encoding originalOutputEncoding = Console.OutputEncoding;
+            Encoding originalInputEncoding = Console.InputEncoding;
+            Thread thisThread = Thread.CurrentThread;
+            CultureInfo originalUICulture = thisThread.CurrentUICulture;
+
+            // Set the UI language based on the SDK environment var.
+            Environment.SetEnvironmentVariable(DOTNET_CLI_UI_LANGUAGE, "ja"); // japanese chose arbitrarily.
+
+            MSBuildApp.SetConsoleUI();
+
+            Assert.Equal(thisThread.CurrentUICulture, new CultureInfo("ja"));
+            Assert.Equal(65001, System.Console.OutputEncoding.CodePage); // utf 8 enabled for correct rendering.
+
+            // Restore the current UI culture back to the way it was at the beginning of this unit test.
+            thisThread.CurrentUICulture = originalUICulture;
+            Environment.SetEnvironmentVariable(DOTNET_CLI_UI_LANGUAGE, originalUILanguage);
+            // MSbuild should also restore the encoding upon exit, but we don't create that context here.
+            Console.OutputEncoding = originalOutputEncoding;
+            Console.InputEncoding = originalInputEncoding;
```

Comment:
This should be done in a try/finally in the case that the test fails (or use the disposable test environment to do that automatically).